pkg/utils/extensions/extensions.go

```diff
@@ -25,49 +25,112 @@ func NewValidator(extensionsMatch, extensionsFilter []string) *Validator {
 	}
 
 	for _, extension := range extensionsMatch {
-		validator.extensionsMatch[normalizeExtension(extension)] = struct{}{}
+		normalized := normalizeExtension(extension)
+		validator.extensionsMatch[normalized] = struct{}{}
+		gologger.Debug().Msgf("Added match extension: %s -> %s", extension, normalized)
 	}
 	for _, item := range defaultDenylist {
-		validator.extensionsFilter[normalizeExtension(item)] = struct{}{}
+		normalized := normalizeExtension(item)
+		validator.extensionsFilter[normalized] = struct{}{}
+		gologger.Debug().Msgf("Added default deny extension: %s -> %s", item, normalized)
 	}
 	for _, extension := range extensionsFilter {
-		validator.extensionsFilter[normalizeExtension(extension)] = struct{}{}
+		normalized := normalizeExtension(extension)
+		validator.extensionsFilter[normalized] = struct{}{}
+		gologger.Debug().Msgf("Added filter extension: %s -> %s", extension, normalized)
 	}
 	return validator
 }
 
-// ValidatePath returns true if an extension is allowed by the validator
-func (e *Validator) ValidatePath(item string) bool {
+// ExactMatch returns true if a path matches the extension match list exactly
+func (e *Validator) ExactMatch(item string) bool {
```

Comment:
i think we should consider using the name `ValidateExtension` instead. it would be more in line with our naming conventions and would also align better with `ValidatePath`, keeping things consistent.